src/js/userProfile.js

```diff
@@ -0,0 +1,162 @@
+import React from "react";
+import API from "@aws-amplify/api";
+import Loading from "./loading";
+import ReactGA from 'react-ga';
+
+const ENDPOINT_URL = process.env.REACT_APP_ENDPOINT_URL;
+
+class UserProfile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userInfo: undefined,
+      dataFetched: false,
+    }
+  }
+
+  async componentDidMount() {
+    // TO DO: Remove authentication for the meet_info API
+    // or create a new API for getting user profiles.
+    const meet_info = await API.get(
+      "treehacks",
+      `/users/${this.props.match.params.userId}/forms/meet_info`,
+      {}
+    );
+    console.log(meet_info);
+    if (meet_info) {
+      this.setState({
+        userInfo: meet_info,
+        dataFetched: true,
+      });
+    }
+  }
+
+  render() {
+    if (!this.state.dataFetched) {
+      return <Loading />;
+    }
+    return (
+      <div className="user-profile">
+        <UserProfileHeader 
+          userInfo={this.state.userInfo} 
+          userId={this.props.match.params.userId}
+        />
+        <UserProfileSummary userInfo={this.state.userInfo} />
+        <UserProfileProject userInfo={this.state.userInfo} />
+        <UserProfileExperience userInfo={this.state.userInfo} />
+        <UserProfileOtherInfo userInfo={this.state.userInfo} />
+      </div>
+    );
+  }
+}
+
+class UserProfileHeader extends React.Component {
```

Comment:
It's better to make these all functional components when possible, especially when they don't need to use state.

Reply:
Fixed!